### PR TITLE
Lesson10 submission for JohnRoge

### PR DESCRIPTION
**Pull request by johnroge**
Pretty sure I'm missing a few opportunities and would appreciate any feedback I can get.

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@johnroge can you please elaborate on "Pretty sure I'm missing a few opportunities" ?

**Comment by johnroge** [comment]
The code runs fine, but I suspect I'm not taking full advantage of any opportunities for things like recursion, list comprehension, using the various methods. Would love to see these things pointed out.
________________________________

File: students/john_rogers/lesson10/donors_fp.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    def total_amount_raised(self):
+        return sum([d.total_donations for d in self.donors])
+
+    def total_number_donations(self):
+        return sum([d.number_of_donations for d in self.donors])
+
+    def thank_all(self):
+        for donor in self.donors:
+            letter = Donor.form_letter(donor)
```
Comment (natasha-aleksandrova) [inline students/john_rogers/lesson10/donors_fp.py]: incorrect usage of class: `form_letter` should be called on an instance.

`donor` is an instance
`Donor` is a class

so should be : `letter = donor.form_letter()`

File: students/john_rogers/lesson10/donors_fp.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            print(letter)
+            print()
+
+    def save_report(self):
+        print('Saving database to disk...')
+        for donor in self.donors:
+            letter = Donor.form_letter(donor)
```
Comment (natasha-aleksandrova) [inline students/john_rogers/lesson10/donors_fp.py]: same here, call the method on an instance

File: students/john_rogers/lesson10/donors_fp.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        """
+        sorted_donors = []
+
+        for donor in self.donors:
+            sorted_donors.append([[donor.full_name],
+                                 [donor.total_donations],
+                                 [donor.number_of_donations],
+                                 [donor.avg_donation_amount]])
+
+        sorted_donors.sort(key=lambda x: x[1])
```
Comment (natasha-aleksandrova) [inline students/john_rogers/lesson10/donors_fp.py]: `sort()` method has a reverse flag that you can set to True

**Comment by johnroge** [comment]
Appreciate the feedback, will get those fixed today.
JR
________________________________

**Pull request merged**